src/Features/CSharp/Portable/Completion/CompletionProviders/ExplicitInterfaceMemberCompletionProvider.cs

```diff
@@ -82,7 +82,7 @@ internal override bool IsInsertionTrigger(SourceText text, int characterPosition
             var members = semanticModel.LookupSymbols(
                 position: name.SpanStart,
                 container: symbol)
-                    .WhereAsArray(s => !s.IsStatic)
+                    .WhereAsArray(s => !s.IsStatic && s.ContainingType.Equals(symbol))
```

Comment:
I'm not certain why we're even using LookupSymbols...  We already have the interface type.  So it's unclear to me why we don't just get the members we need directly off that type.